lib/client/rpc/openbookMarkets.ts

```diff
@@ -8,66 +8,63 @@ import {
   OpenOrders,
   SelfTradeBehaviorUtils,
   PlaceOrderTypeUtils,
+  OPENBOOK_PROGRAM_ID,
 } from "@openbook-dex/openbook-v2";
 import numeral from "numeral";
 import {
   Transaction,
   PublicKey,
   TransactionInstruction,
 } from "@solana/web3.js";
-import { getAssociatedTokenAddressSync } from "@solana/spl-token";
+import {
+  TOKEN_PROGRAM_ID,
+  getAssociatedTokenAddressSync,
+} from "@solana/spl-token";
 import {
   OpenbookMarket,
+  OpenbookMarketFetchRequest,
   OpenbookOrder,
   OrderBookSideType,
   Orderbook,
   PlaceOrderType,
   ProposalAccountWithKey,
   TwapPlaceOrderArgs,
-  TwapProgram,
 } from "@/types";
 import { FutarchyMarketsClient } from "@/client";
 import { TransactionSender } from "@/transactions";
-import { ProgramVersion } from "@/types";
 import { enrichTokenMetadata } from "@/tokens";
 import { getTwapMarketKey } from "@/openbookTwap";
-import {
-  BASE_FORMAT,
-  NUMERAL_FORMAT,
-  autocratVersionToTwapMap,
-} from "@/constants";
+import { BASE_FORMAT, MAX_MARKET_PRICE, NUMERAL_FORMAT } from "@/constants";
```

Comment:
I mentioned this in an earlier post, but these formatting rules will likely need to be dynamic and more robust. Currently from what I recall they are quite naive..... For example if we have a market where the price can go to 6 decimals (we should display those where it's relevant (trading)) but also some understanding of the values too.

For example if the token price is 0.00017 then we can't just simply round and display truncated 0.0.

This is a future note, but will be important for our value formatting which will be a lift from the UI.

Reply:
duly noted. Yes we need to be in a better state indeed